pyfstat/make_sfts.py

```diff
@@ -129,10 +146,92 @@ def basic_setup(self):
+        If noiseSFTs are given, use them to obtain relevant data parameters (tstart,
+        duration, detectors and Tsft). The corresponding input values will be used to
+        set up a lalpulsar.SFTConstraints object to be imposed to the SFTs. Keep in
+        mind that Tsft will also be checked to be consistent accross all SFTs (this is
+        not implemented through SFTConstraints but through a simple list check).
+        """
+        SFTConstraint = self._get_sft_constraints_from_tstart_duration()
+        noise_multi_sft_catalog = lalpulsar.GetMultiSFTCatalogView(
+            lalpulsar.SFTdataFind(self.noiseSFTs, SFTConstraint)
+        )
+
+        # Information to be extracted from the SFTs themselves
+        IFOs = []
+        tstart = []
+        tend = []
+        Tsft = []
+        self.sftfilenames = []  # This refers to the MFD output!
+
+        # SWIG-LAL does not import this functionality
+        gps_to_int = lambda x: x.gpsSeconds
+
+        for ifo_catalog in noise_multi_sft_catalog.data:
+            ifo_name = lalpulsar.ListIFOsInCatalog(ifo_catalog).data[0]
+
+            time_stamps = lalpulsar.TimestampsFromSFTCatalog(ifo_catalog)
+            this_Tsft = int(round(1.0 / ifo_catalog.data[0].header.deltaF))
+            this_start_time = gps_to_int(time_stamps.data[0])
+            this_end_time = gps_to_int(time_stamps.data[-1]) + this_Tsft
+
+            self.sftfilenames.append(
+                lalpulsar.OfficialSFTFilename(
+                    ifo_name[0],
+                    ifo_name[1],
+                    ifo_catalog.length,
+                    this_Tsft,
+                    this_start_time,
+                    this_end_time - this_start_time,
+                    self.label,
+                )
+            )
+
+            IFOs.append(ifo_name)
+            tstart.append(this_start_time)
+            tend.append(this_end_time)
+            Tsft.append(this_Tsft)
+
+        # Get the "overall" values of the search
+        Tsft = np.unique(Tsft)
+        if len(Tsft) != 1:
+            raise ValueError("SFTs contain different basetimes: {}".format(Tsft))
+        elif Tsft[0] != self.Tsft:
+            raise ValueError(
+                "SFT basetime {} differs from input base time {}".format(
+                    Tsft[0], self.Tsft
+                )
+            )
+        self.tstart = min(tstart)
+        self.tend = max(tend)
+        self.duration = self.tend - self.tstart
+        self.IFOs = ",".join(['"{}"'.format(d) for d in IFOs])
+
+    def basic_setup(self):
+        os.makedirs(self.outdir, exist_ok=True)
+        self.config_file_name = os.path.join(self.outdir, self.label + ".cff")
+        self.theta = np.array([self.phi, self.F0, self.F1, self.F2])
+
+        if self.noiseSFTs is not None:
+            logging.warning(
+                "noiseSFTs is not None: Inferring tstart, duration, tend, Tsft. "
+                "Input tstart, duration, Tsft and detectors will be treated "
```

Comment:
Technically Tsft is now not used in the constraint, but it is sanity-checked, so I guess this is ok.